lib/beaker/hypervisor/docker.rb

```diff
@@ -64,11 +64,25 @@ def provision
           image_name = image.id
         end
 
-        @logger.debug("Creating container from image #{image_name}")
-        container = ::Docker::Container.create({
+        container = []
+        container_opts = {
           'Image' => image_name,
           'Hostname' => host.name,
-        })
+        }
+
+        unless host['docker_container_name'].nil?
```

Comment:
I think the logic here could be a little bit better.

- Not using the 'container' variable since its re-used later, rename into something else.
- Also the container = [] should be renamed. all kinds of weird assignments are happening here.
- container.any? should only be done if we actually set the docker_container_name ( otherwise follow normal workflow )